record-editor.py

```diff
@@ -212,15 +195,88 @@ def commit_record( self ):
 
         Returns nothing.
         """
+        # notify any listeners of the ID of the record committed.
+        self.committed.emit( self.record["id"] )
+
+    def closeEvent( self, event ):
+        """
+        Handles closing the window by calling the callback specified at window
+        creation.
+
+        Takes 1 argument:
+
+          event - QCloseEvent used to accept or reject the close event.
+
+        Returns nothing.
+        """
+        # notify any listeners that this RecordEditor has been closed.
+        self.closed.emit( self.record["id"] )
+
+# XXX: refactor into widgets.py?
+class SelectionView( QTableView ):
+    """
 
-        # run our callback if we have one.
-        if self.commit_callback is not None:
-            self.commit_callback()
+    """
+    def __init__( self, parent=None ):
+        """
+
+        """
+        super().__init__( parent )
+
+        self.setSelectionBehavior( QAbstractItemView.SelectRows )
+        self.setSelectionMode( QAbstractItemView.SingleSelection )
+        self.setEditTriggers( QAbstractItemView.NoEditTriggers )
+        self.setAlternatingRowColors( True )
+        self.verticalHeader().hide()
+        self.verticalHeader().setDefaultSectionSize( 20 )
+        self.setSortingEnabled( True )
+
+        # prevent the users from rearranging the columns.
+        self.horizontalHeader().setSectionsMovable( False )
+
+        # have the last column fill the remaining space of the table
+        # XXX: for some reason in ArtRecordViewer this doesn't take
+        #      until you manually adjust the horizontal header,
+        #      another weird QMainWindow side effect?
+        self.horizontalHeader().setStretchLastSection( True )
+
+    def sizeHint( self ):
+        """
+        Provides a size suitable for viewing the table with all its columns
+        resized to their contents.
+
+        Takes no arguments.
+
+        Return 1 value:
+
+          QSize - containing the suggested width and height.
+
+        """
+        default = super().sizeHint()
+
+        if self.model():
+            self.resizeColumnsToContents()
+
+            width = sum( [self.columnWidth( i )
+                          for i in range( self.model().columnCount() )] )
+
+            if self.verticalScrollBar().isVisible():
+                width += self.verticalScrollBar().width()
```

Comment:
If I add 12 or more to this value then the horizontal scroll bar disappears due to sizeHint() matching the actual width.  A cursory search doesn't enlighten me as to why as what you have here looks correct.  Does the horizontal scroll bar appear by default for you?